### PR TITLE
[ML] Make task_type optional

### PR DESCRIPTION
**Pull request by davidkyle**
Makes the `task_type` element of the _inference API optional so that it is now possible to GET, DELETE or POST to an inference entity without stating the task type up front. 

The new endpoints are:
```
DELETE /_inference/<model_id>
GET /_inference/<model_id>
PUT /_inference/<model_id>
POST /_inference/<model_id>
```
When the `PUT /_inference/<model_id>` form is used `task_type` must be specified in the body

### CODE REVIEW

File: docs/changelog/104483.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 104483
+summary: Make `task_type` optional
```
Comment (droberts195) [inline docs/changelog/104483.yaml:2]:
```suggestion
summary: Make `task_type` optional in `_inference` APIs
```

**Review by jonathan-buttner — APPROVED** [review]
Looks good. Do we need a PR to update the client code as well?

**Review by szabosteve — APPROVED** [review]
Docs LGTM! Thanks!

**Comment by davidkyle** [comment]
> Looks good. Do we need a PR to update the client code as well?

The clients will pick up the new paths from the .json spec files

**Pull request merged**